tools/go-agent/Makefile

```diff
@@ -23,12 +23,15 @@ BINARY = skywalking-go-agent
 SH = sh
 GO = go
 GIT = git
+GIT_COMMIT := $(shell $(GIT) rev-parse --short HEAD)
+GO_VERSION := $(shell $(GO) version | sed 's/go version go\([0-9.]*\).*/\1/')
+VERSION_PACKAGE := main
 PROTOC = protoc
 GO_PATH = $$($(GO) env GOPATH)
 GO_BUILD = $(GO) build
 GO_LINT = $(GO_PATH)/bin/golangci-lint
 GO_BUILD_FLAGS = -v
-GO_BUILD_LDFLAGS = -X main.version=$(VERSION)
+GO_BUILD_LDFLAGS = -X $(VERSION_PACKAGE).version=$(VERSION) -X $(VERSION_PACKAGE).goVersion=$(GO_VERSION) -X $(VERSION_PACKAGE).gitCommit=$(GIT_COMMIT)
```

Comment:
I have concern of doing this. Note, in the ASF, the release tar means the one on the SVN. Not the github one.

Reply:
Don't worry, this only captures environment information during compilation and will not be interfered by other factors

Reply:
I am not saying this could cause an error. I am saying this would generate a different binary tar when users build from the official release. This should not happen.

All builds of official releases and binaries should be reproducible.

Reply:
Sorry, I misunderstood you.

When users refer to [build-docs](https://skywalking.apache.org/docs/skywalking-go/latest/en/development-and-contribution/build-and-use-agent/#build-the-agent) to build, in fact The obtained version is also Unknown
```shell
$ cd skywalking-go && make build
...

$ cd ./bin/ && ./skywalking-go-agent--windows-amd64 -version
skywalking-go agent version: unknown
```
In sw-go,`release.sh` actually controls the version:
https://github.com/apache/skywalking-go/blob/7948b433093a6ff51a7131a8f86ecfe5fd9ed56c/tools/release/create_bin_release.sh#L26-L38

Of course, I also support this after revision.  If the user uses `make build` directly, the output will be as follows:
```shell
$ cd skywalking-go && make build
...

$ cd ./bin/ && ./skywalking-go-agent--windows-amd64 -version
skywalking-go agent
 Version: Unknown
 Go version: go1.22.2
 Git commit: a6824f6
```

I'm not sure if I understand you correctly, thanks for your reply

Reply:
Then it was not correct.
The version and commit ID should be a part of source tar, to make sure it could be repackaged from the source.

Reply:
@mrproliu I am not sure what the way of current go agent. But the policy should be, if we have version and commit ID in the binary in the release we built, users should have that when they build from our release(source tar in ASF).

Reply:
Right. I have talked with @ShyunnY, He will save the version information as a file and save it to the release package. Then, we can resolve this issue.

Reply:
I insert the VERSION file when compiling src tgz, so that the information obtained by users directly downloading the source package is always consistent with the information obtained by git clone. 

This presents the idea that "no matter how you install, the information is always consistent"